### PR TITLE
BENCH: Benchmarks for interpolate module

### PR DESCRIPTION
**Pull request by maniteja123**
The following functions are being benchmarked:
- [x] interp1d
- [x] interp2d
- [x] Rbf
- [x] griddata
- [x] UnivariateSpline
- [x] SmoothBivariateSpline
- [x] LSQBivariateSpline

### CODE REVIEW

**Comment by maniteja123** [comment]
I had added the example @josef-pkt provided on the mailing list for the bivariate spline for benchmarking. It would be great if the domain specialists comment on this, as I am no expert on this topic :)

**Comment by ev-br** [comment]
Can you add the link to the mailing list thread with the bivariate spline please.

Also a nice bonus could be a comparative benchmark of `interp1d(..., kind='linaer')` and `np.interp` (I suspect the latter is a bit more performant)

**Comment by maniteja123** [comment]
I think this is a [working link](http://scipy-user.10969.n7.nabble.com/BivariateSpline-examples-and-my-crashing-python-td14801.html). 

Regarding the comparitive benchmark, thanks for the information :). I would love to do that. On an additional note, is there any PR or issue in which the better performance of ``np.interp`` was discussed in particular so that I could try to understand it better ?

File: benchmarks/benchmarks/interpolate.py [hunk 1]
```diff
@@ -64,10 +64,132 @@ def setup(self):
...
+
+    def time_univariate_spline(self, n_samples, degree):
+        interpolate.UnivariateSpline(self.x, self.y, k=degree)
+
+
+class BivariateSpline(Benchmark):
+    """
+    Created on Thu Nov 26 22:00:20 2009
+    Author: josef-pktd and scipy mailinglist example
+    """
```
Comment (ev-br) [inline benchmarks/benchmarks/interpolate.py]: Ideally, the mailing list message is mentioned here, in the docstring or a code comment. Rationale: source code is more robust than github issues.

**Comment by ev-br** [comment]
> is there any PR or issue in which the better performance of np.interp was discussed in particular so that I could try to understand it better ?

Not to my knowledge, no.

File: benchmarks/benchmarks/interpolate.py (hunk 1 above)
Comment (maniteja123) [inline benchmarks/benchmarks/interpolate.py]: Got the point. Will follow it from now on.

**Comment by maniteja123** [comment]
Okay, no problem. I will try to understand as much as I can if there is any particular pros and cons for each of them and then do it sometime in this week.

**Comment by maniteja123** [comment]
I have added a benchmark comparing ``np.interp`` and ``interp1d``. Also, the hyperlink in the docstring has exceeded the limit of 79 characters. Please inform if that is okay.

**Comment by ev-br** [comment]
Okay, appartently,  the difference between `np.interp` and `scipy.interpolate1d(..., kind='linear')` is only by a factor of about five for these small sizes and unsorted input.

**Comment by ev-br** [comment]
At some point I'd like to see also a comparison for the sorted input as well. But this can be done separately, unless tweaking the benchmarks after the fact is this is going to mess up the nightly benchmarking Pauli runs. @pv?

**Comment by pv** [comment]
Changes in the benchmark won't automatically invalidate the old result;
you need to invalidate result manually in case there are substantial
changes on what is measured.

**Comment by ev-br** [comment]
Thanks for clarification @pv, will add the sorted input comparison as a separate benchmark then. 
Merging, thank you @maniteja123

**Pull request merged**